internal/config/config.go

```diff
@@ -62,26 +70,30 @@ type Manifest struct {
 	YAML     string
 }
 
-// TestCase represents a test-case to be run by kuttl.
+// TestCase represents a test-case to be run by chainsaw.
 type TestCase struct {
-	Timeout            int
+	Timeout            time.Duration
 	SetupScriptPath    string
 	TeardownScriptPath string
 	SkipUpdate         bool
 	SkipImport         bool
 
 	OnlyCleanUptestResources bool
+
+	TestDirectory string
 }
 
 // Resource represents a Kubernetes object to be tested and asserted
 // by uptest.
 type Resource struct {
-	Name      string
-	Namespace string
-	KindGroup string
-	YAML      string
-
-	Timeout              int
+	Name       string
+	Namespace  string
+	KindGroup  string
+	YAML       string
+	APIVersion string
+	Kind       string
```

Comment:
Yes this is duplicate. We need to access `Kind` and `KindGroup` in template files. I did not move the complexity to template file for doing concatenation inside the template files. Instead of this, I introduced both of them as separate fields.